gdal/ogr/ogrsf_frmts/shape/shpopen.c

```diff
@@ -504,21 +497,21 @@ SHPOpen( const char * pszLayer, const char * pszAccess )
 /*                                                                      */
 /*      Open the .shp and .shx files based on the basename of the       */
 /*      files or either file name.                                      */
+/*      There are 2 types of SHPOpen() : with .shx restoring code       */
+/*      (SHPOpenLLEx) and normal one.                                   */
```

Comment:
We afraid to change shapelib ABI, so decided not to change SHPOpenLL signature, but add new function with extended parameters. But if changing SHPOpenLL is normal, it can be easily fixed.